### DISCUSSION TITLE
How to send a large amount of data via MQTT?

### DISCUSSION THREAD

**Discussion by ZergOfSwarm** [cesanta/mongoose#2715] [Ask questions, get answers] [answered]
I need to send a large amount of data via MQTT, up to 2048 bytes! Which function should I use, mg_mqtt_pub() or publish_status()? Do I need to increase any buffers in the library itself? 

And the second question, how quickly can I send data? For example, can I send 2KB every 1ms?

**Comment by scaprile** [comment, marked as answer]
I don't see any publish_status() function in the Mongoose API
I don't see any example using other function than mg_mqtt_pub()
Please follow the guidelines in the [documentation](https://mongoose.ws/documentation/),  [examples](https://github.com/cesanta/mongoose/tree/master/examples/) and [tutorials](https://mongoose.ws/documentation/#tutorials-and-examples) available.

publish_status() is not a Mongoose API function but just an example of a function that _publish_ es the _status_ of something in an example of a user application. Guess what function it is using:
https://github.com/cesanta/mongoose/blob/c7f9fe4a03132cfc33b278cebe6484b78f2e7807/examples/mqtt-dashboard/device/net.c#L83-L110

Please read https://mongoose.ws/documentation/#best-practices

We don't have timing limitations in Mongoose, and we don't support multithreading, so there is no way you could send at a faster speed than your loop lets you send. If you refer to the poll manager timeout, it can be  as short or as large as your other actions in the same loop allow
